core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Driver.java

```diff
@@ -142,37 +136,32 @@ public String encode(Barcodable barcodable, int copies) {
     public String encode(Barcodable barcodable, int copies) {
       StringBuilder sb = new StringBuilder();
 
-      try {
-        String barcode = getBarcode64(barcodable);
-        String alias = barcodable.getLabelText();
-        String name = barcodable.getName();
-        String barcode64 = new String(Base64.encodeBase64(barcode.getBytes("UTF-8")));
-        sb.append("m m\n");
-        sb.append("J\n");
-        sb.append("S l1;0,0,12,15,38\n");
-        sb.append("B 3,2,0,DATAMATRIX,0.21;").append(barcode64).append("\n");
-        sb.append("B 17,1,0,DATAMATRIX+RECT,0.25;").append(barcode64).append("\n");
-        sb.append("T 29,2,0,5,pt4;[DATE]\n");
-        appendTruncated(17, name, s -> appendBradyEscapedUnicode(sb, s));
-        sb.append("T 17,8,0,5,pt6;");
-        appendBradyEscapedUnicode(sb, alias);
-        sb.append("\n");
-        sb.append("T 17,11,0,5,pt6;");
-        appendBradyEscapedUnicode(sb, name);
-        sb.append("\n");
-        sb.append("A 1").append("\n");
-        appendTruncated(17, alias, s -> appendBradyEscapedUnicode(sb, s));
-        sb.append("T 17,8,0,5,pt6;");
-        appendBradyEscapedUnicode(sb, alias);
-        sb.append("\n");
-        sb.append("T 17,11,0,5,pt6;");
-        appendBradyEscapedUnicode(sb, name);
-        sb.append("\n");
-        sb.append("A ").append(copies).append("\n");
-      } catch (UnsupportedEncodingException e) {
-        log.error("get raw state", e);
-        return null;
-      }
+      String barcode = getBarcode64(barcodable);
+      String alias = barcodable.getLabelText();
+      String name = barcodable.getName();
+      String barcode64 = new String(Base64.encodeBase64(barcode.getBytes(StandardCharsets.UTF_8)));
+      sb.append("m m\n");
+      sb.append("J\n");
+      sb.append("S l1;0,0,12,15,38\n");
+      sb.append("B 3,2,0,DATAMATRIX,0.21;").append(barcode64).append("\n");
+      sb.append("B 17,1,0,DATAMATRIX+RECT,0.25;").append(barcode64).append("\n");
+      sb.append("T 29,2,0,5,pt4;[DATE]\n");
+      appendTruncated(17, name, s -> appendBradyEscapedUnicode(sb, s));
+      sb.append("T 17,8,0,5,pt6;");
+      appendBradyEscapedUnicode(sb, alias);
+      sb.append("\n");
+      sb.append("T 17,11,0,5,pt6;");
+      appendBradyEscapedUnicode(sb, name);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "appendBradyEscapedUnicode()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)